porch/engine/pkg/engine/engine.go

```diff
@@ -102,6 +103,15 @@ func (cad *cadEngine) CreatePackageRevision(ctx context.Context, repositoryObj *
 		mutations = append(mutations, mutation)
 	}
 
+	// If creating a package in a deployment repository, generate context
+	if repositoryObj.Spec.Deployment {
```

Comment:
One observation: `kpt CLI`'s [implementation](https://github.com/GoogleContainerTools/kpt/blob/main/internal/util/get/get.go#L125) is deviating significantly from this server side logic.

May be not a huge pain at the moment, but can become an issue as we evolve the deployment hook more.

Reply:
Can we change kpt to create library of the 'customize for deployment' functionality, similarly to what we did with render (it will need to accept function runner for anything other than builtins but this could be a better solution. How hard would it be to refactor kpt to do that?

Reply:
we have a package `hook` under [internal/hook](https://github.com/GoogleContainerTools/kpt/blob/main/internal/hook/executor.go#L60) which exposes the this as a library (It also get's invoked in `kpt pkg get` command as library only`). 

So might be able to use it with little changes.